framework/db/ActiveQueryTrait.php

```diff
@@ -43,6 +48,18 @@ public function asArray($value = true)
     }
 
     /**
+     * Sets the [[collectionClass]] property.
+     * @param string $className collection class to be instantiated and filled with results.
+     * @return $this the query object itself
+     * @since 2.0.10
+     */
+    public function asCollection($className)
```

Comment:
See https://github.com/yiisoft/yii2/pull/12304#issuecomment-242328482

$className should be optional here, while default collection class is used.